packages/did-provider-ebsi/src/services/EbsiRPCService.ts

```diff
@@ -0,0 +1,157 @@
+  AddVerificationMethodRelationshipParams,
+  GetDidDocumentParams,
+  GetDidDocumentsParams,
+  InsertDidDocumentParams,
+  SendSignedTransactionParams,
+  UpdateBaseDocumentParams,
+  Response,
+  GetDidDocumentsResponse,
+} from '../types'
+import { DIDDocument } from 'did-resolver'
+
+/**
+ * @constant {string} jsonrpc
+ */
+const jsonrpc = '2.0' // optional param of plugin?
+const baseUrl = 'https://api-pilot.ebsi.eu/did-registry/v5/jsonrpc' // optional param of plugin?
+const token = '' // optional param of plugin?
+
+/**
+ * Call to build an unsigned transaction to insert a new DID document. Requires an access token with "didr_invite" or
+ * "didr_write" scope.
+ * @param {{ id: InsertDidDocumentParams[], id: number }} args
+ */
+export const insertDidDocument = async (args: { params: InsertDidDocumentParams[]; id: number }): Promise<Response> => {
+  const { params, id } = args
+  const options = {
+    method: 'POST',
+    headers: new Headers({
+      Authorization: 'Bearer ' + token,
+    }),
+    body: JSON.stringify({
+      jsonrpc,
+      method: 'insertDidDocument',
+      params,
+      id,
+    }),
+  }
+  return await (await fetch(baseUrl, options)).json()
+}
+
+/**
+ * Call to build an unsigned transaction to update the base document of an existing DID. Requires an access token with
+ * "didr_write" scope.
+ * @param {{ params: UpdateBaseDocumentParams[], id:number }} args
+ */
+export const updateBaseDocument = async (args: { params: UpdateBaseDocumentParams[]; id: number }): Promise<Response> => {
+  const { params, id } = args
+  const options = {
+    method: 'POST',
+    headers: new Headers({
+      Authorization: 'Bearer ' + token,
+    }),
+    body: JSON.stringify({
+      jsonrpc,
+      method: 'updateBaseDocument',
+      params,
+      id,
+    }),
+  }
+  return await (await fetch(baseUrl, options)).json()
+}
+
+/**
+ * Call to build an unsigned transaction to add a verification method. Requires an access token with "didr_write" scope.
+ * @param {{ params: AddVerificationMethodParams[], id:number }} args
+ */
+export const addVerificationMethod = async (args: { params: AddVerificationMethodParams[]; id: number }): Promise<Response> => {
+  const { params, id } = args
+  const options = {
+    method: 'POST',
+    headers: new Headers({
+      Authorization: 'Bearer ' + token,
+    }),
+    body: JSON.stringify({
+      jsonrpc,
+      method: 'addVerificationMethod',
+      params,
+      id,
+    }),
+  }
+  return await (await fetch(baseUrl, options)).json()
+}
+
+/**
+ * Call to build an unsigned transaction to add a verification relationship. Requires an access token with "didr_write" scope.
+ * @param {{ params: AddVerificationMethodRelationshipParams[], id: number }} args
+ */
+export const addVerificationMethodRelationship = async (args: {
+  params: AddVerificationMethodRelationshipParams[]
+  id: number
+}): Promise<Response> => {
+  const { params, id } = args
+  const options = {
```

Comment:
Please extract these into a method. We now have the same things all over the place, with only a different method. If we need to change something in the future the chance of introducing bugs is high

Reply:
moved the code to the `buildFetchOptions(...)` method